pkg/tfgen/docs.go

```diff
@@ -822,82 +821,166 @@ func (p *tfMarkdownParser) parseSchemaWithNestedSections(subsection []string) {
+	regexp.MustCompile("###+ ([a-z_0-9]+).*"),
 
 	// For example:
 	// dynamodb_table.html.markdown: "### `server_side_encryption`"
-	regexp.MustCompile("###+ `([a-z_]+).*`"),
+	regexp.MustCompile("###+ `([a-z_0-9]+).*`"),
 
 	// For example:
 	// route53_record.html.markdown: "### Failover Routing Policy"
-	regexp.MustCompile("###+ ([a-zA-Z_ ]+).*"),
+	// appflow_flow.html.markdown: "###### S3 Input Format Config"
+	regexp.MustCompile("###+ ([a-zA-Z_ 0-9]+).*"),
 
 	// For example:
 	// sql_database_instance.html.markdown:
 	// "The optional `settings.ip_configuration.authorized_networks[]`` sublist supports:"
 	regexp.MustCompile("`([a-zA-Z_.\\[\\]]+)`.*supports:"),
+
+	// For example when blocks/subblocks/sublists are defined across more than one line
+	// sql_database_instance.html.markdown:
+	//"The optional `settings.maintenance_window` subblock for instances declares a one-hour"
+	regexp.MustCompile("The .* `([a-zA-Z_.\\[\\]]+)` sublist .*"),
+	regexp.MustCompile("The .* `([a-zA-Z_.\\[\\]]+)` subblock .*"),
+	regexp.MustCompile("The .* `([a-zA-Z_.\\[\\]]+)` block .*"),
+}
+
+// getMultipleNestedBlockNames is called when we detect that a resource matches the "`([a-z_0-9]+)`.*following" regex.
+// We check if more than one nested block name is listed on this line, and we additionally check if there's an
+// indication of an extra nested object, denoted by "'s", as in:
+// "The `grpc_route`, `http_route` and `http2_route` 's `action` object supports the following:".
+func getMultipleNestedBlockNames(match string) []string {
+	subNest := ""
+	var nestedBlockNames []string
+	// First we check for the presence of a possible nested property via 's
+	if strings.Contains(match, "'s ") {
+		// split the match along the 's
+		part1, part2, _ := strings.Cut(match, "'s")
+		match = part1
+		// Extract our subheading. It should be the second item in part2 of the match.
+		part2Slice := strings.Split(part2, "`")
+		if len(part2Slice) >= 2 {
+			subNest = part2Slice[1]
+		}
+	}
+	// As per previous regex match, the resource names will be surrounded by backticks, so we extract them
+	tokenInBackticks := regexp.MustCompile("`[^`]+`")
+	blockNames := tokenInBackticks.FindAllString(match, -1)
+	for _, blockName := range blockNames {
+		if blockName != "" {
+			blockName = strings.Trim(blockName, "`")
+			blockName = strings.ToLower(blockName)
+			blockName = strings.Replace(blockName, " ", "_", -1)
+			blockName = strings.TrimSuffix(blockName, "[]")
+			if subNest != "" {
+				// For the format ""The `grpc_route`, `http_route` and `http2_route` 's `action` object
+				//supports the following:" the result should be grpc_route.action
+				blockName = blockName + "." + subNest
+			}
+			nestedBlockNames = append(nestedBlockNames, blockName)
+		}
+	}
+	return nestedBlockNames
 }
 
-// getNestedBlockName take a line of a Terraform docs Markdown page and returns the name of the nested block it
+func splitMatchOnColon(match string) []string {
```

Comment:
I'm not sure I understand what this function does? Can we add an example in the doc comment and maybe look at changing the name. To me, `splitMatchOnColon` implies `strings.Split(match, ":")`.